lib/app.js

```diff
@@ -0,0 +1,112 @@
+var clone = require('clone');
+
+function App(appref, id, fbRef, data, _firebase, _newBlock) {
```

Comment:
this is one part I'm on the fence about. Constructor args? We never use more than this. Options object? Cleaner code... marginally.